### DISCUSSION TITLE
Prometheus query editor: prettify query

### DISCUSSION THREAD

**Discussion by dimitarvdimitrov** [grafana/grafana#68579] [Data Sources]
Prometheus supports the [`/api/v1/format_query` endpoint](https://prometheus.io/docs/prometheus/latest/querying/api/#formatting-query-expressions) which takes a promql expression and prettifies it. 

Recently this was added to Grafana Mimir too. See https://github.com/grafana/mimir/pull/4373

Grafana can use this endpoint to format the query in the editor.

**Comment by gtk-grafana** [comment]
Looks like support was added in [prometheus 2.38](https://prometheus.io/docs/prometheus/2.38/querying/api/#formatting-query-expressions), and cortex/thanos [shortly after](https://github.com/cortexproject/cortex/commit/05e905b9fc0798e448f62a2b893f319c2d351a2e#diff-c46c9cc11bd167b16ba8100564bd05a2c1174eac0983f0288d4be02fad144433).

I'll get this in the backlog, this could be a quick win for the code editor.
Thank you for the heads up @dimitarvdimitrov

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This has an issue, #68732 and further discussion should go there. Thank you!
